### PR TITLE
Feat search video

### PR DESCRIPTION
**Pull request by amartya-rs**
In this PR: 
- search videos in explore page

### CODE REVIEW

File: src/components/top-navigation/TopNav.jsx [hunk 1]
```diff
@@ -1,21 +1,41 @@
...
+         <div className="search-bar">
+            <input
+               className="text-input"
+               type="text"
+               placeholder="Search"
+               value={searchInput}
+               onChange={(e) => {
+                  setSearchInput(e.target.value.toLowerCase());
+                  pathname !== "/explore" && navigate("/explore");
+               }}
```
Comment (zoef-yt) [inline src/components/top-navigation/TopNav.jsx]: extract it into a function and call that function here.

**Review by sudipkundu999 — APPROVED** [review]
The code looks good!
Good to merge.

**Review by zoef-yt — APPROVED** [review]
Small pr, suggested a change. UI looks great. the functionality works as intended. You are good to merge.

**Review by saurabhan — APPROVED** [review]
Code works fine and puts out expected behavior,
Just a suggestion: make the search only filter when 2 or more characters are added, as a single character almost results in every single item.

File: src/components/top-navigation/TopNav.jsx (hunk 1 above)
Comment (amartya-rs) [reply src/components/top-navigation/TopNav.jsx]: Done.

**Comment by amartya-rs** [comment]
> Code works fine and puts out expected behavior, Just a suggestion: make the search only filter when 2 or more characters are added, as a single character almost results in every single item.

Thanks for pointing this out, will work on it.

**Pull request merged**